### PR TITLE
added a fix for the bug that stops the site, added functionality to f…

### PR DESCRIPTION
**Pull request by isabelleh**
I added a few lines in bundle.js so that when it hits an empty xml set it just gets a new one instead of checking for tags, because the tag check was causing it to get stuck. 

For filtering based on a comment, I added to the DoDrawWay function. It uses the code from the show comment function to get the information. It is a little repetitive. Also, its unclear whether it is a useful addition, because it has to check the change sets individually. Theoretically it could be useful during a large edit session if the majority of edits taken from overpass where from the session, but it has not been tested at such an event.

### CODE REVIEW

**Comment by isabelleh** [comment]
Also not sure why it shows me "changing" that last 100 lines or so because they are the same, so no need to look at that.. I can replace them with the original code to be sure there was nothing accidentally changed.

**Comment by iandees** [comment]
Thanks for sending this in, @isabelleh! I'll take a look. @tmcw if you have some time while you're on the road it would help to have your input, too.

File: js/bundle.js [hunk 1]
```diff
@@ -31,7 +39,7 @@ var overview_map = L.map('overview_map', {
 
 var bing = new L.BingLayer(BING_KEY, 'Aerial').addTo(map);
 
-var osm = new L.TileLayer('//a.tiles.mapbox.com/v3/saman.map-f8nluy8d/{z}/{x}/{y}.jpg70', {
+var osm = new L.TileLayer('http://a.tiles.mapbox.com/v3/saman.map-f8nluy8d/{z}/{x}/{y}.jpg70', {
```
Comment (tmcw) [inline js/bundle.js]: These urls are purposely started with `//` in order to use https when available

**Comment by tmcw** [comment]
Looks fine to me but the replacement of protocol-relative urls with http:// urls is problematic - if we're going to hardcode a protocol, it should be https.

**Comment by isabelleh** [comment]
Ok I  can replace those. I had to switch them to be able to run locally while testing

File: index.html [hunk 2]
```diff
@@ -1,13 +1,13 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <title>Show Me The Way</title>
-    <meta charset='utf-8' />
-    <link rel='stylesheet' href='//cdnjs.cloudflare.com/ajax/libs/leaflet/0.5/leaflet.css' />
+    <title>Show me the way</title>
+    <meta nrset='utf-8' />
```
Comment (tmcw) [inline index.html]: Why is this changed?

File: index.html [hunk 3]
```diff
@@ -41,7 +41,7 @@
 
       (function() {
         var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
-        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
+        ga.src = 'js/ga.js';
```
Comment (tmcw) [inline index.html]: there's no ga.js file added in this branch.

File: index.html [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-    <meta charset='utf-8' />
-    <link rel='stylesheet' href='//cdnjs.cloudflare.com/ajax/libs/leaflet/0.5/leaflet.css' />
+    <title>Show me the way</title>
+    <meta nrset='utf-8' />
+    <link rel='stylesheet' href='css/leaflet.css' />
     <link rel='stylesheet' href='css/style.css' />
     <link rel='stylesheet' href='css/site.css' />
     <!--[if lte IE 8]><link rel='stylesheet' href='http://cdn.leafletjs.com/leaflet-0.5/leaflet.ie.css' /><![endif]-->
-    <link href='//fonts.googleapis.com/css?family=Open+Sans:400,700|Open+Sans+Condensed:700' rel='stylesheet' type='text/css'>
+    <link href='css/fonts.css' rel='stylesheet' type='text/css'>
```
Comment (tmcw) [inline index.html]: there's no fonts.css file added in this branch.

File: index.html [hunk 5]
```diff
@@ -1,13 +1,13 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <title>Show Me The Way</title>
-    <meta charset='utf-8' />
-    <link rel='stylesheet' href='//cdnjs.cloudflare.com/ajax/libs/leaflet/0.5/leaflet.css' />
+    <title>Show me the way</title>
+    <meta nrset='utf-8' />
+    <link rel='stylesheet' href='css/leaflet.css' />
```
Comment (tmcw) [inline index.html]: this file is not present in this branch.

File: index.html [hunk 6]
```diff
@@ -30,7 +30,7 @@
      <%= past_tense[change.type] %> <a target='_blank' href='//osm.org/browse/<%= change.meta.type %>/<%= change.meta.id %>'><%= change.tagtext %></a>
     </p>
     </script>
-    <script src='//cdnjs.cloudflare.com/ajax/libs/leaflet/0.5/leaflet.js'></script>
+    <script src='js/leaflet.js'></script>
```
Comment (tmcw) [inline index.html]: are these files added locally?

File: index.html (hunk 3 above)
Comment (isabelleh) [inline index.html]: Yes. Thank you! I forgot about the changes that I had to make to be able to run locally and have no switched all of those back

**Comment by iandees** [comment]
Thanks for your contribution, @isabelleh!

**Comment by tmcw** [comment]
Thanks @isabelleh !
